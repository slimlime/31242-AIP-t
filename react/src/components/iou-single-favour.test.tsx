import React from "react";
import { shallow } from "enzyme";
import IouFavour from "./iou-single-favour";

describe("<IouFavour />", () => {
  it("should render correctly", () => {
    const wrapper = shallow(
      <IouFavour
        giverDisplayName="James Long Lee"
        recieverDisplayName="Kevin Pog Lueng"
        item={{ id: "1", display_name: "Coffee" }}
      />
    );
    expect(wrapper.find("#favourAvatar").at(0).text()).toEqual("JL");
    expect(wrapper.find("#favourAvatar").at(1).text()).toEqual("KL");
    expect(wrapper.find("#favourAvatarText").at(0).text()).toEqual(
      "James Long Lee"
    );
    expect(wrapper.find("#favourAvatarText").at(1).text()).toEqual(
      "Kevin Pog Lueng"
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should render correctly with no reciever", () => {
    const wrapper = shallow(
      <IouFavour
        giverDisplayName="James Long Lee"
        recieverDisplayName=""
        item={{ id: "1", display_name: "Coffee" }}
      />
    );
    expect(wrapper.find("#favourAvatar").at(0).text()).toEqual("JL");
    expect(wrapper.find("#favourAvatar").at(1).text()).toEqual("");
    expect(wrapper.find("#favourAvatarText").at(0).text()).toEqual(
      "James Long Lee"
    );
    expect(wrapper.find("#favourAvatarText").at(1).text()).toEqual(
      "No Reciever"
    );
  });
});
