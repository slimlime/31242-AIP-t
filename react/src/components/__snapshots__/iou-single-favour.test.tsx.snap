// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IouFavour /> should render correctly 1`] = `
<div
  id="requestItem"
>
  <div
    id="favourAvatarContainer"
  >
    <WithStyles(ForwardRef(Avatar))
      id="favourAvatar"
      onClick={[Function]}
    >
      JL
    </WithStyles(ForwardRef(Avatar))>
    <WithStyles(ForwardRef(Popover))
      anchorEl={null}
      anchorOrigin={
        Object {
          "horizontal": "center",
          "vertical": "bottom",
        }
      }
      id="popUpMargin"
      onClose={[Function]}
      open={false}
      transformOrigin={
        Object {
          "horizontal": "center",
          "vertical": "top",
        }
      }
    >
      <WithStyles(ForwardRef(Typography))
        id="favourAvatarText"
      >
        James Long Lee
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Popover))>
  </div>
  <KeyboardArrowRightIcon
    color="primary"
    id="favourArrow"
  />
  <div
    id="rewardContainer"
  >
    <WithStyles(ForwardRef(Avatar))
      id="rewardAvatar"
    >
      Coffee
    </WithStyles(ForwardRef(Avatar))>
  </div>
  <KeyboardArrowRightIcon
    color="primary"
    id="favourArrow"
  />
  <div
    id="favourAvatarContainer"
  >
    <WithStyles(ForwardRef(Avatar))
      id="favourAvatar"
      onClick={[Function]}
    >
      KL
    </WithStyles(ForwardRef(Avatar))>
    <WithStyles(ForwardRef(Popover))
      anchorEl={null}
      anchorOrigin={
        Object {
          "horizontal": "center",
          "vertical": "bottom",
        }
      }
      id="popUpMargin"
      onClose={[Function]}
      open={false}
      transformOrigin={
        Object {
          "horizontal": "center",
          "vertical": "top",
        }
      }
    >
      <WithStyles(ForwardRef(Typography))
        id="favourAvatarText"
      >
        Kevin Pog Lueng
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Popover))>
  </div>
</div>
`;
